File: modules/x-engine-native-broadcast/android/native-broadcast/src/main/java/com/zkty/nativ/broadcast/NativeBroadcast.java
package com.zkty.nativ.broadcast;

import android.util.Log;

import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.jsi.webview.XEngineWebView;
import com.zkty.nativ.jsi.webview.XWebViewPool;

import java.util.HashMap;
import java.util.Map;

public class NativeBroadcast extends NativeModule implements IBroadcast {

    @Override
    public String moduleId() {
        return "com.zkty.native.broadcast";
    }

    @Override
    public void broadcast(String type, String payload) {

        Map<String, String> bro = new HashMap<>();
        bro.put("type", type);
        bro.put("payload", payload);

        for (XEngineWebView webView : XWebViewPool.sharedInstance().getWebViews()) {
            webView.callHandler("com.zkty.module.engine.broadcast", new Object[]{bro}, retValue -> Log.d("NativeBroadcast", "broadcast:" + payload));
        }
        for (XEngineWebView webView : XWebViewPool.sharedInstance().getTabWebViewList()) {
            webView.callHandler("com.zkty.module.engine.broadcast", new Object[]{bro}, retValue -> Log.d("NativeBroadcast", "broadcast:" + payload));
        }
    }
}

File: modules/x-engine-native-broadcast/iOS/Class/Native_broadcast.m
//
//  Native_broadcast.m
//  broadcast
//
//  Created by zk on 2020/9/7.
//  Copyright © 2020 x-engine. All rights reserved.


#import "Native_broadcast.h"
#import "XENativeContext.h"
#import "WebViewFactory.h"
#import "XEngineWebView.h"

@interface Native_broadcast()
{ }
@end

@implementation Native_broadcast
NATIVE_MODULE(Native_broadcast)

 - (NSString*) moduleId{
    return @"com.zkty.native.broadcast";
}

- (int) order{
    return 0;
}

- (void)afterAllNativeModuleInited{
} 

-(void) broadcast:(NSString*) type payload:(NSString*) payload{
    for (XEngineWebView* wv in [WebViewFactory sharedInstance].webviews){
        if(wv){
            [wv callHandler:@"com.zkty.module.engine.broadcast" arguments:@{
                @"type":type,
                @"payload":payload
            }
             completionHandler:^(id  _Nullable value) {
                NSLog(@"js broadcast ret value %@",value);
            }];
        }
    }
}
@end
 
